package/MDAnalysis/core/topologyattrs.py

```diff
@@ -1799,6 +1800,137 @@ def n_fragments(self):
+          that all atoms of these molecules belong the group.
+
+        See Also
+        --------
+        :func:`MDAnalysis.lib.mdamath.make_whole`
+
+
+        .. versionadded:: 0.20.0
+        """
+        if reference is not None:
+            if reference.lower() not in ['com', 'cog']:
+                raise ValueError("Unrecognized reference definition: {}\n"
+                                 "Please use one of 'com', 'cog', or None."
+                                 "".format(compound))
+            ref = reference.lower()
+
+        if box is None:
+            box = self.dimensions
+        boxtype, box = check_box(box)
+        ortho = (boxtype == 'ortho')
+        if ortho:
+            hbox = 0.5 * box[:3]
+
+        atoms = self.atoms
+        unique_atoms = atoms.unique
+
+        comp = compound.lower()
+        if comp == 'group':
+            if ortho:
+                positions = unique_atoms.positions
+                # Only unwrap if necessary:
+                if np.any(np.abs(positions - positions[0]) > hbox):
+                    positions = mdamath.make_whole(unique_atoms, inplace=False)
+            else:
+                positions = mdamath.make_whole(unique_atoms, inplace=False)
+            # Apply reference shift if required:
+            if reference is not None:
+                if ref == 'com':
+                    refpos = unique_atoms.center_of_mass(pbc=False,
+                                                         compound='group')
+                elif ref == 'cog':
+                    refpos = unique_atoms.center_of_geometry(pbc=False,
+                                                             compound='group')
+                target = apply_PBC(refpos, self.dimensions)
+                positions += target - refpos
+            if inplace:
+                unique_atoms.positions = positions
+            if not atoms.isunique:
+                positions = positions[atoms._unique_restore_mask]
+            return positions
+        elif comp == 'fragments':
+            compound_indices = unique_atoms.fragindices
+        elif comp == 'residues':
+            compound_indices = unique_atoms.resindices
+        elif comp == 'segments':
+            compound_indices = unique_atoms.segindices
+        elif comp == 'molecules':
+            try:
+                compound_indices = unique_atoms.molnums
+            except AttributeError:
+                raise NoDataError("Cannot use compound='molecules': "
+                                  "No molecule information in topology.")
+        else:
+            raise ValueError("Unrecognized compound definition: {}\nPlease use"
+                             " one of 'group', 'residues', 'segments', "
+                             "'molecules', or 'fragments'.".format(compound))
+
+        # Now process every compound:
+        unique_compound_indices = unique_int_1d(compound_indices)
```

Comment:
I don't like the structuring here, you've got `comp == group` being returned before this, then this says it's where the work is done.  Maybe instead:
```python
if not comp in ALLOWED:
   ValueError:
elif comp == 'group'
    positions = simple unpack
else:
    # find compound indices
    if comp == 'residue':
        # etc
    elif comp == 'fragment':
        # etc
    positions = loop over compounds

return positions
```

So there's a single exit point?